skyportal/handlers/api/shift.py

```diff
@@ -478,12 +475,14 @@ def patch(self, shift_id, user_id):
             shiftuser.admin = admin
 
             needs_replacement = data.get("needs_replacement", False)
-            try:
-                needs_replacement = bool(needs_replacement)
-            except (ValueError, TypeError):
-                return self.error(
-                    "Invalid needs_replacement parameter: unable to parse to boolean"
-                )
+            # test if bool already
+            if type(needs_replacement) != bool:
+                try:
+                    needs_replacement = bool(needs_replacement)
+                except (ValueError, TypeError):
+                    return self.error(
+                        "Invalid needs_replacement parameter: unable to parse to boolean"
+                    )
```

Comment:
If it is a boolean you don't need any of this try/except.

Reply:
good point